lib/services/redfish-obm-service.js

```diff
@@ -0,0 +1,145 @@
+    };
+
+    redfishObmService.prototype.reboot = function() {
+        if (!this.params.force) {
+            return this._runInternal('GracefulRestart');
+        }
+        return this._runInternal('ForceRestart');
+    };
+    
+    redfishObmService.prototype.NMI = function() {
+        return this._runInternal('Nmi');
+    };
+    
+    redfishObmService.prototype.powerButton = function() {
+        return this._runInternal('PushPowerButton');
+    };
+    
+    redfishObmService.prototype.powerStatus = function() {
+        return Promise.resolve(); // (jl) TODO
+    };
+        
+    redfishObmService.prototype._initClient = function () {
+        var apiClient = new redfish.ApiClient();
+        apiClient.basePath = this.options.config.uri.replace(/\/+$/, '');
+        
+        // setup basic authorization
+        if (!_.isUndefined(this.config.username) && 
+            !_.isUndefined(this.config.password)) {
+            var token = new Buffer(
+                this.config.username + ':' + this.config.password
+            ).toString('base64');
+            apiClient.defaultHeaders.Authorization = 'Basic ' + token;
+            process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0";
+        }
+        
+        var redfishApi = Promise.promisifyAll(new redfish.RedfishvApi(apiClient));
+        return redfishApi;
+    };
+    
+    redfishObmService.prototype._checkAction = function(action) {
+        return this.redfishApi.listResetTypesAsync(this.targetId)
+        .then(function(res) {
+            var resetTypes = res[1].body['reset_type@Redfish.AllowableValues'];
+            assert.ok(Array.isArray(resetTypes), 
+                'expected an array of reset types');
+            if (0 > _.indexOf(resetTypes, action)) {
+                throw new Error('unsupported reset type ' + action);
+            }
+        })
+        .then(function() {
+            return Promise.resolve();
+        })
+        .catch(function(err) {
+            return Promise.reject(err);
+        });
+    };
+    
+    redfishObmService.prototype._runInternal = function (action) {
+        return this.run({
+            action:action
+        });
+    };
+    
+    redfishObmService.prototype.run = function (options) {       
+        var self = this;
+        assert.object(options);
+        assert.string(options.action);
+        var action = options.action;
+        return self._checkAction(action)
+        .then(function() {
+            return self.redfishApi.doResetAsync(
+                self.targetId, 
+                { reset_type: action }
+            )
+            .catch(function(err) {
+                throw new Error(err.response.text);
+            });
+        })
+        .then(function(res) {
+            assert.ok(Array.isArray(res), 'expected an array response type');
+            var taskId = res[1].body['@odata.id'];
+            assert.string(taskId);
+            return Promise.resolve(taskId);
```

Comment:
You can just `return taskId` since it will get promisified if returned from within a then block.